### DISCUSSION TITLE
Thoughts on a Microsoft-owned resolver or on cooperation with the .NET team?

### DISCUSSION THREAD

**Discussion by antonfirsov** [MichaCo/DnsClient.NET#210] [Ideas]
I'm a member of the .NET Networking team looking at options to address https://github.com/dotnet/runtime/issues/19443 in some form, hopefully in the .NET 9 timeframe. I'd like to get feedback on my ideas. Note that nothing is official yet, at this point it's just me gathering information.

Long story short, there seems to be increased interest having a resolver lib with at least some Microsoft support behind it. IMO we have two options to achieve this: 

1. Team up to co-maintain DnsClient.NET
2. Develop a separate library (maybe in BCL), potentially using insights (in some cases maybe code) form DnsClient.NET

I see several pros and cons for both options, may elaborate if there is interest.

@MichaCo any thoughts/feelings/preferences regarding (1) and (2)? Are you open to participate in discussions around this topic? May I contact you offline?

**Comment by MichaConrad** [comment]
Hi @antonfirsov ,
Yes, that sounds very interesting and I am absolutely interested in discussions around that topic.
Lets discuss offline!
